File: Airline Reservation System/frontend/src/services/FlightService.js
import axios from "axios";

const BASE_URL = "http://localhost:1515/api/v1/flights";

class FlightService {

    saveFlight(flight) {
        return axios.post(BASE_URL, flight);
    }

    findAllFlights() {
        return axios.get(BASE_URL);
    }

    findFlightById(id) {
        return axios.get(BASE_URL + "/" + id);
    }

    updateFlight(id, flight) {
        return axios.put(BASE_URL + "/" + id, flight);
    }

    deleteFlight(id) {
        return axios.delete(BASE_URL + "/" + id);
    }

    updateFlightStatus(id, status) {
        return axios.put(BASE_URL + "/" + id + "/status/" + status);
    }

    searchFlights(departureAirportId, arrivalAirportId, journeyDate) {
        return axios.get(
            BASE_URL +
            "/search?departureAirportId=" + departureAirportId +
            "&arrivalAirportId=" + arrivalAirportId +
            "&journeyDate=" + journeyDate
        );
    }

}

export default new FlightService();

File: Airline Reservation System/backend/src/main/java/com/coforge/ars/service/FlightService.java
package com.coforge.ars.service;

import java.time.LocalDateTime;
import java.util.List;

import java.util.Optional;

import com.coforge.ars.model.Flight;



public interface FlightService {
	boolean saveFlight(Flight flight);
	boolean updateFlight(Integer flightId,Flight flight);
	boolean deleteFlightById(Integer flightId);
	Optional<Flight> findByFlightId(Integer flightId);
	List<Flight> findAllFlights();
	
	//to update flight status
	boolean updateFlightStatus(Integer flightId,String flightStatus);
	
	List<Flight> searchFlights(Integer departureAirportId,
			Integer arrivalAirportId,
			LocalDateTime journeyDate);
}
